Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrBridgeBuilder.kt

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin.native.interop.gen
+
+import org.jetbrains.kotlin.native.interop.gen.jvm.KotlinPlatform
+import org.jetbrains.kotlin.native.interop.indexer.ObjCProtocol
+import org.jetbrains.kotlin.utils.addToStdlib.firstIsInstanceOrNull
+
+class BridgeBuilderResult(
+        val kotlinFile: KotlinFile,
+        val nativeBridges: NativeBridges,
+        val propertyAccessorBridgeBodies: Map<PropertyAccessor, String>,
+        val functionBridgeBodies: Map<FunctionStub, List<String>>
+)
+
+/**
+ * Generates [NativeBridges] and corresponding function bodies and property accessors.
+ */
+class StubIrBridgeBuilder(
+        private val context: StubIrContext,
+        private val builderResult: StubIrBuilderResult) {
+
+    private val globalAddressExpressions = mutableMapOf<String, KotlinExpression>()
+
+    private fun getGlobalAddressExpression(cGlobalName: String, accessor: PropertyAccessor) =
+            globalAddressExpressions.getOrPut(cGlobalName) {
+                simpleBridgeGenerator.kotlinToNative(
```

Comment:
It is somewhat incorrect to register only the first requesting accessor as `nativeBacked` for this stub. `globalAddressExpressions` should probably have both function parameters in its keys.